// @rmwc/chip/next/trailing-action/foundation.js
import { useFoundation } from '@rmwc/base';
import { MDCChipTrailingActionFoundation } from '@material/chips';
export var useTrailingActionFoundation = function (props) {
    var foundationWithElements = useFoundation({
        props: props,
        elements: { rootEl: true },
        foundation: function (_a) {
            var rootEl = _a.rootEl, emit = _a.emit;
            return new MDCChipTrailingActionFoundation({
                focus: function () { var _a; return (_a = rootEl.ref) === null || _a === void 0 ? void 0 : _a.focus(); },
                getAttribute: function (attr) {
                    return rootEl.ref && rootEl.ref.getAttribute(attr);
                },
                notifyInteraction: function (trigger) {
                    emit('onInteraction', { trigger: trigger });
                },
                notifyNavigation: function (key) {
                    emit('onNavigation', { key: key });
                },
                setAttribute: function (attr, value) {
                    return rootEl.setProp(attr, value);
                }
            });
        }
    });
    return foundationWithElements;
};
