// @fluentui/react-skeleton/lib-commonjs/components/Skeleton/useSkeleton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useSkeleton_unstable = void 0;
const react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");
const SkeletonContext_1 = /*#__PURE__*/require("../../contexts/SkeletonContext");
/**
 * Create the state required to render Skeleton.
 *
 * The returned state can be modified with hooks such as useSkeletonStyles_unstable,
 * before being passed to renderSkeleton_unstable.
 *
 * @param props - props from this instance of Skeleton
 * @param ref - reference to root HTMLElement of Skeleton
 */
const useSkeleton_unstable = (props, ref) => {
  const {
    animation: contextAnimation,
    appearance: contextAppearance
  } = SkeletonContext_1.useSkeletonContext();
  const {
    animation = contextAnimation !== null && contextAnimation !== void 0 ? contextAnimation : 'wave',
    appearance = contextAppearance !== null && contextAppearance !== void 0 ? contextAppearance : 'opaque'
  } = props;
  const root = react_utilities_1.getNativeElementProps('div', {
    ref,
    role: 'progressbar',
    'aria-busy': true,
    'aria-label': 'Loading Content',
    ...props
  });
  return {
    animation,
    appearance,
    components: {
      root: 'div'
    },
    root
  };
};
exports.useSkeleton_unstable = useSkeleton_unstable;
