// @fluentui/react-card/lib-amd/components/CardHeader/useCardHeader.js
define(["require", "exports", "tslib", "react", "@fluentui/react-utilities", "../Card/CardContext", "./useCardHeaderStyles"], function (require, exports, tslib_1, React, react_utilities_1, CardContext_1, useCardHeaderStyles_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useCardHeader_unstable = void 0;
    /**
     * Create the state required to render CardHeader.
     *
     * The returned state can be modified with hooks such as useCardHeaderStyles_unstable,
     * before being passed to renderCardHeader_unstable.
     *
     * @param props - props from this instance of CardHeader
     * @param ref - reference to root HTMLElement of CardHeader
     */
    var useCardHeader_unstable = function (props, ref) {
        var image = props.image, header = props.header, description = props.description, action = props.action;
        var _a = CardContext_1.useCardContext_unstable().selectableA11yProps, referenceId = _a.referenceId, setReferenceId = _a.setReferenceId;
        var headerRef = React.useRef(null);
        var generatedId = react_utilities_1.useId(useCardHeaderStyles_1.cardHeaderClassNames.header, referenceId);
        React.useEffect(function () {
            if (header && headerRef.current) {
                var id = headerRef.current.id;
                setReferenceId(id ? id : generatedId);
            }
        }, [header, setReferenceId, generatedId]);
        return {
            components: {
                root: 'div',
                image: 'div',
                header: 'div',
                description: 'div',
                action: 'div',
            },
            root: react_utilities_1.getNativeElementProps('div', tslib_1.__assign({ ref: ref }, props)),
            image: react_utilities_1.resolveShorthand(image),
            header: react_utilities_1.resolveShorthand(header, {
                required: true,
                defaultProps: {
                    ref: headerRef,
                    id: referenceId,
                },
            }),
            description: react_utilities_1.resolveShorthand(description),
            action: react_utilities_1.resolveShorthand(action),
        };
    };
    exports.useCardHeader_unstable = useCardHeader_unstable;
});
