// @fluentui/react-card/lib-amd/components/CardPreview/useCardPreviewStyles.js
define(["require", "exports", "@griffel/react"], function (require, exports, react_1) {
    "use strict";
    var _a;
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useCardPreviewStyles_unstable = exports.cardPreviewClassNames = void 0;
    /**
     * Static CSS class names used internally for the component slots.
     */
    exports.cardPreviewClassNames = {
        root: 'fui-CardPreview',
        logo: 'fui-CardPreview__logo',
    };
    var useStyles = react_1.makeStyles({
        root: (_a = {
                position: 'relative'
            },
            _a["> :not(." + exports.cardPreviewClassNames.logo + ")"] = {
                display: 'block',
                height: '100%',
                width: '100%',
            },
            _a),
        logo: {
            position: 'absolute',
            bottom: '12px',
            left: '12px',
            width: '32px',
            height: '32px',
        },
    });
    /**
     * Apply styling to the CardPreview slots based on the state.
     */
    var useCardPreviewStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.cardPreviewClassNames.root, styles.root, state.root.className);
        if (state.logo) {
            state.logo.className = react_1.mergeClasses(exports.cardPreviewClassNames.logo, styles.logo, state.logo.className);
        }
        return state;
    };
    exports.useCardPreviewStyles_unstable = useCardPreviewStyles_unstable;
});
