// @rmwc/linear-progress/next/foundation.js
import { __assign, __read, __rest } from "tslib";
import { useEffect, useState } from 'react';
import { MDCLinearProgressFoundation } from '@material/linear-progress';
import { useFoundation } from '@rmwc/base';
export var useLinearProgressFoundation = function (props) {
    var setStyleWithSelector = function (ref, selector, styleProperty, value) {
        var _a;
        var el = (_a = ref === null || ref === void 0 ? void 0 : ref.querySelector(selector)) !== null && _a !== void 0 ? _a : null;
        el && (el.style[styleProperty] = value);
    };
    var _a = useFoundation({
        props: props,
        elements: { rootEl: true },
        foundation: function (_a) {
            var rootEl = _a.rootEl;
            return new MDCLinearProgressFoundation({
                addClass: function (className) { return rootEl.addClass(className); },
                forceLayout: function () { var _a; return (_a = rootEl.ref) === null || _a === void 0 ? void 0 : _a.offsetWidth; },
                hasClass: function (className) { return rootEl.hasClass(className); },
                removeClass: function (className) { return rootEl.removeClass(className); },
                setBufferBarStyle: function (styleProperty, value) {
                    setStyleWithSelector(rootEl.ref, MDCLinearProgressFoundation.strings.BUFFER_BAR_SELECTOR, styleProperty, value);
                },
                setPrimaryBarStyle: function (styleProperty, value) {
                    setStyleWithSelector(rootEl.ref, MDCLinearProgressFoundation.strings.PRIMARY_BAR_SELECTOR, styleProperty, value);
                },
                setAttribute: function (name, value) {
                    rootEl.setProp(name, value);
                },
                removeAttribute: function (name) { return rootEl.removeProp(name); }
            });
        }
    }), foundation = _a.foundation, elements = __rest(_a, ["foundation"]);
    var _b = __read(useState(undefined), 2), determinate = _b[0], setDeterminate = _b[1];
    // progress and determinate
    useEffect(function () {
        foundation.setProgress(props.progress || 0);
        var isDeterminate = props.progress !== undefined;
        if (isDeterminate !== determinate) {
            foundation.setDeterminate(isDeterminate);
            setDeterminate(isDeterminate);
        }
    }, [props.progress, determinate, foundation]);
    // buffer
    useEffect(function () {
        foundation.setBuffer(props.buffer || 0);
    }, [props.buffer, foundation]);
    // reversed
    useEffect(function () {
        foundation.setReverse(!!props.reversed);
    }, [props.reversed, foundation]);
    // closed
    useEffect(function () {
        props.closed ? foundation.close() : foundation.open();
    }, [props.closed, foundation]);
    return __assign({ foundation: foundation }, elements);
};
