// @fluentui/react-components/lib-commonjs/unstable/index.js
"use strict";

// Stub for unstable exports
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useVirtualizer_unstable = exports.virtualizerClassNames = exports.Virtualizer = exports.useSkeletonContext = exports.SkeletonContextProvider = exports.useSkeletonItem_unstable = exports.useSkeletonItemStyles_unstable = exports.skeletonItemClassNames = exports.renderSkeletonItem_unstable = exports.SkeletonItem = exports.useSkeleton_unstable = exports.useSkeletonStyles_unstable = exports.skeletonClassNames = exports.renderSkeleton_unstable = exports.Skeleton = exports.useField_unstable = exports.useFieldStyles_unstable = exports.renderField_unstable = exports.fieldClassNames = exports.Field = exports.textareaFieldClassNames = exports.TextareaField = exports.switchFieldClassNames = exports.SwitchField = exports.spinButtonFieldClassNames = exports.SpinButtonField = exports.sliderFieldClassNames = exports.SliderField = exports.selectFieldClassNames = exports.SelectField = exports.radioGroupFieldClassNames = exports.RadioGroupField = exports.progressFieldClassNames = exports.ProgressField = exports.inputFieldClassNames = exports.InputField = exports.comboboxFieldClassNames = exports.ComboboxField = exports.checkboxFieldClassNames = exports.CheckboxField = exports.renderInfoButton_unstable = exports.useInfoButtonStyles_unstable = exports.useInfoButton_unstable = exports.infoButtonClassNames = exports.InfoButton = exports.useAlert_unstable = exports.useAlertStyles_unstable = exports.renderAlert_unstable = exports.alertClassNames = exports.Alert = void 0;
exports.flattenTree_unstable = exports.useFlatTree_unstable = exports.treeClassNames = exports.treeItemClassNames = exports.treeItemLayoutClassNames = exports.treeItemLevelToken = exports.treeItemPersonaLayoutClassNames = exports.useTreeItemLayoutStyles_unstable = exports.useTreeItemPersonaLayoutStyles_unstable = exports.useTreeItemStyles_unstable = exports.useTreeStyles_unstable = exports.renderTreeItemLayout_unstable = exports.renderTreeItemPersonaLayout_unstable = exports.renderTreeItem_unstable = exports.renderTree_unstable = exports.useTreeItemPersonaLayout_unstable = exports.useTreeItemLayout_unstable = exports.useTreeItem_unstable = exports.useTree_unstable = exports.useTreeItemContext_unstable = exports.useTreeContext_unstable = exports.TreeItemPersonaLayout = exports.TreeItemProvider = exports.TreeItemLayout = exports.TreeProvider = exports.TreeItem = exports.Tree = exports.useIntersectionObserver = exports.useVirtualizerStyles_unstable = exports.renderVirtualizer_unstable = void 0;
var react_alert_1 = /*#__PURE__*/require("@fluentui/react-alert");
Object.defineProperty(exports, "Alert", {
  enumerable: true,
  get: function () {
    return react_alert_1.Alert;
  }
});
Object.defineProperty(exports, "alertClassNames", {
  enumerable: true,
  get: function () {
    return react_alert_1.alertClassNames;
  }
});
Object.defineProperty(exports, "renderAlert_unstable", {
  enumerable: true,
  get: function () {
    return react_alert_1.renderAlert_unstable;
  }
});
Object.defineProperty(exports, "useAlertStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_alert_1.useAlertStyles_unstable;
  }
});
Object.defineProperty(exports, "useAlert_unstable", {
  enumerable: true,
  get: function () {
    return react_alert_1.useAlert_unstable;
  }
});
var react_infobutton_1 = /*#__PURE__*/require("@fluentui/react-infobutton");
Object.defineProperty(exports, "InfoButton", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.InfoButton;
  }
});
Object.defineProperty(exports, "infoButtonClassNames", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.infoButtonClassNames;
  }
});
Object.defineProperty(exports, "useInfoButton_unstable", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.useInfoButton_unstable;
  }
});
Object.defineProperty(exports, "useInfoButtonStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.useInfoButtonStyles_unstable;
  }
});
Object.defineProperty(exports, "renderInfoButton_unstable", {
  enumerable: true,
  get: function () {
    return react_infobutton_1.renderInfoButton_unstable;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_checkbox_1 = /*#__PURE__*/require("@fluentui/react-checkbox");
Object.defineProperty(exports, "CheckboxField", {
  enumerable: true,
  get: function () {
    return react_checkbox_1.CheckboxField_unstable;
  }
});
Object.defineProperty(exports, "checkboxFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_checkbox_1.checkboxFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_combobox_1 = /*#__PURE__*/require("@fluentui/react-combobox");
Object.defineProperty(exports, "ComboboxField", {
  enumerable: true,
  get: function () {
    return react_combobox_1.ComboboxField_unstable;
  }
});
Object.defineProperty(exports, "comboboxFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_combobox_1.comboboxFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_input_1 = /*#__PURE__*/require("@fluentui/react-input");
Object.defineProperty(exports, "InputField", {
  enumerable: true,
  get: function () {
    return react_input_1.InputField_unstable;
  }
});
Object.defineProperty(exports, "inputFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_input_1.inputFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_progress_1 = /*#__PURE__*/require("@fluentui/react-progress");
Object.defineProperty(exports, "ProgressField", {
  enumerable: true,
  get: function () {
    return react_progress_1.ProgressField_unstable;
  }
});
Object.defineProperty(exports, "progressFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_progress_1.progressFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_radio_1 = /*#__PURE__*/require("@fluentui/react-radio");
Object.defineProperty(exports, "RadioGroupField", {
  enumerable: true,
  get: function () {
    return react_radio_1.RadioGroupField_unstable;
  }
});
Object.defineProperty(exports, "radioGroupFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_radio_1.radioGroupFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_select_1 = /*#__PURE__*/require("@fluentui/react-select");
Object.defineProperty(exports, "SelectField", {
  enumerable: true,
  get: function () {
    return react_select_1.SelectField_unstable;
  }
});
Object.defineProperty(exports, "selectFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_select_1.selectFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_slider_1 = /*#__PURE__*/require("@fluentui/react-slider");
Object.defineProperty(exports, "SliderField", {
  enumerable: true,
  get: function () {
    return react_slider_1.SliderField_unstable;
  }
});
Object.defineProperty(exports, "sliderFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_slider_1.sliderFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_spinbutton_1 = /*#__PURE__*/require("@fluentui/react-spinbutton");
Object.defineProperty(exports, "SpinButtonField", {
  enumerable: true,
  get: function () {
    return react_spinbutton_1.SpinButtonField_unstable;
  }
});
Object.defineProperty(exports, "spinButtonFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_spinbutton_1.spinButtonFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_switch_1 = /*#__PURE__*/require("@fluentui/react-switch");
Object.defineProperty(exports, "SwitchField", {
  enumerable: true,
  get: function () {
    return react_switch_1.SwitchField_unstable;
  }
});
Object.defineProperty(exports, "switchFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_switch_1.switchFieldClassNames;
  }
});
// eslint-disable-next-line deprecation/deprecation
var react_textarea_1 = /*#__PURE__*/require("@fluentui/react-textarea");
Object.defineProperty(exports, "TextareaField", {
  enumerable: true,
  get: function () {
    return react_textarea_1.TextareaField_unstable;
  }
});
Object.defineProperty(exports, "textareaFieldClassNames", {
  enumerable: true,
  get: function () {
    return react_textarea_1.textareaFieldClassNames;
  }
});
var react_field_1 = /*#__PURE__*/require("@fluentui/react-field");
Object.defineProperty(exports, "Field", {
  enumerable: true,
  get: function () {
    return react_field_1.Field;
  }
});
Object.defineProperty(exports, "fieldClassNames", {
  enumerable: true,
  get: function () {
    return react_field_1.fieldClassNames;
  }
});
Object.defineProperty(exports, "renderField_unstable", {
  enumerable: true,
  get: function () {
    return react_field_1.renderField_unstable;
  }
});
Object.defineProperty(exports, "useFieldStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_field_1.useFieldStyles_unstable;
  }
});
Object.defineProperty(exports, "useField_unstable", {
  enumerable: true,
  get: function () {
    return react_field_1.useField_unstable;
  }
});
var react_skeleton_1 = /*#__PURE__*/require("@fluentui/react-skeleton");
Object.defineProperty(exports, "Skeleton", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.Skeleton;
  }
});
Object.defineProperty(exports, "renderSkeleton_unstable", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.renderSkeleton_unstable;
  }
});
Object.defineProperty(exports, "skeletonClassNames", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.skeletonClassNames;
  }
});
Object.defineProperty(exports, "useSkeletonStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.useSkeletonStyles_unstable;
  }
});
Object.defineProperty(exports, "useSkeleton_unstable", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.useSkeleton_unstable;
  }
});
Object.defineProperty(exports, "SkeletonItem", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.SkeletonItem;
  }
});
Object.defineProperty(exports, "renderSkeletonItem_unstable", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.renderSkeletonItem_unstable;
  }
});
Object.defineProperty(exports, "skeletonItemClassNames", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.skeletonItemClassNames;
  }
});
Object.defineProperty(exports, "useSkeletonItemStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.useSkeletonItemStyles_unstable;
  }
});
Object.defineProperty(exports, "useSkeletonItem_unstable", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.useSkeletonItem_unstable;
  }
});
Object.defineProperty(exports, "SkeletonContextProvider", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.SkeletonContextProvider;
  }
});
Object.defineProperty(exports, "useSkeletonContext", {
  enumerable: true,
  get: function () {
    return react_skeleton_1.useSkeletonContext;
  }
});
var react_virtualizer_1 = /*#__PURE__*/require("@fluentui/react-virtualizer");
Object.defineProperty(exports, "Virtualizer", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.Virtualizer;
  }
});
Object.defineProperty(exports, "virtualizerClassNames", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.virtualizerClassNames;
  }
});
Object.defineProperty(exports, "useVirtualizer_unstable", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.useVirtualizer_unstable;
  }
});
Object.defineProperty(exports, "renderVirtualizer_unstable", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.renderVirtualizer_unstable;
  }
});
Object.defineProperty(exports, "useVirtualizerStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.useVirtualizerStyles_unstable;
  }
});
Object.defineProperty(exports, "useIntersectionObserver", {
  enumerable: true,
  get: function () {
    return react_virtualizer_1.useIntersectionObserver;
  }
});
var react_tree_1 = /*#__PURE__*/require("@fluentui/react-tree");
Object.defineProperty(exports, "Tree", {
  enumerable: true,
  get: function () {
    return react_tree_1.Tree;
  }
});
Object.defineProperty(exports, "TreeItem", {
  enumerable: true,
  get: function () {
    return react_tree_1.TreeItem;
  }
});
Object.defineProperty(exports, "TreeProvider", {
  enumerable: true,
  get: function () {
    return react_tree_1.TreeProvider;
  }
});
Object.defineProperty(exports, "TreeItemLayout", {
  enumerable: true,
  get: function () {
    return react_tree_1.TreeItemLayout;
  }
});
Object.defineProperty(exports, "TreeItemProvider", {
  enumerable: true,
  get: function () {
    return react_tree_1.TreeItemProvider;
  }
});
Object.defineProperty(exports, "TreeItemPersonaLayout", {
  enumerable: true,
  get: function () {
    return react_tree_1.TreeItemPersonaLayout;
  }
});
Object.defineProperty(exports, "useTreeContext_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeContext_unstable;
  }
});
Object.defineProperty(exports, "useTreeItemContext_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItemContext_unstable;
  }
});
Object.defineProperty(exports, "useTree_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTree_unstable;
  }
});
Object.defineProperty(exports, "useTreeItem_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItem_unstable;
  }
});
Object.defineProperty(exports, "useTreeItemLayout_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItemLayout_unstable;
  }
});
Object.defineProperty(exports, "useTreeItemPersonaLayout_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItemPersonaLayout_unstable;
  }
});
Object.defineProperty(exports, "renderTree_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.renderTree_unstable;
  }
});
Object.defineProperty(exports, "renderTreeItem_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.renderTreeItem_unstable;
  }
});
Object.defineProperty(exports, "renderTreeItemPersonaLayout_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.renderTreeItemPersonaLayout_unstable;
  }
});
Object.defineProperty(exports, "renderTreeItemLayout_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.renderTreeItemLayout_unstable;
  }
});
Object.defineProperty(exports, "useTreeStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeStyles_unstable;
  }
});
Object.defineProperty(exports, "useTreeItemStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItemStyles_unstable;
  }
});
Object.defineProperty(exports, "useTreeItemPersonaLayoutStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItemPersonaLayoutStyles_unstable;
  }
});
Object.defineProperty(exports, "useTreeItemLayoutStyles_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useTreeItemLayoutStyles_unstable;
  }
});
Object.defineProperty(exports, "treeItemPersonaLayoutClassNames", {
  enumerable: true,
  get: function () {
    return react_tree_1.treeItemPersonaLayoutClassNames;
  }
});
Object.defineProperty(exports, "treeItemLevelToken", {
  enumerable: true,
  get: function () {
    return react_tree_1.treeItemLevelToken;
  }
});
Object.defineProperty(exports, "treeItemLayoutClassNames", {
  enumerable: true,
  get: function () {
    return react_tree_1.treeItemLayoutClassNames;
  }
});
Object.defineProperty(exports, "treeItemClassNames", {
  enumerable: true,
  get: function () {
    return react_tree_1.treeItemClassNames;
  }
});
Object.defineProperty(exports, "treeClassNames", {
  enumerable: true,
  get: function () {
    return react_tree_1.treeClassNames;
  }
});
Object.defineProperty(exports, "useFlatTree_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.useFlatTree_unstable;
  }
});
Object.defineProperty(exports, "flattenTree_unstable", {
  enumerable: true,
  get: function () {
    return react_tree_1.flattenTree_unstable;
  }
});
