// @fluentui/react-card/lib-amd/components/CardFooter/useCardFooterStyles.js
define(["require", "exports", "tslib", "@griffel/react"], function (require, exports, tslib_1, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useCardFooterStyles_unstable = exports.cardFooterClassNames = void 0;
    /**
     * Static CSS class names used internally for the component slots.
     */
    exports.cardFooterClassNames = {
        root: 'fui-CardFooter',
        action: 'fui-CardFooter__action',
    };
    var useStyles = react_1.makeStyles({
        root: tslib_1.__assign({ display: 'flex', flexDirection: 'row' }, react_1.shorthands.gap('12px')),
        action: {
            marginLeft: 'auto',
        },
    });
    /**
     * Apply styling to the CardFooter slots based on the state.
     */
    var useCardFooterStyles_unstable = function (state) {
        var styles = useStyles();
        state.root.className = react_1.mergeClasses(exports.cardFooterClassNames.root, styles.root, state.root.className);
        if (state.action) {
            state.action.className = react_1.mergeClasses(exports.cardFooterClassNames.action, styles.action, state.action.className);
        }
        return state;
    };
    exports.useCardFooterStyles_unstable = useCardFooterStyles_unstable;
});
