// @fluentui/react-skeleton/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useSkeletonContext = exports.SkeletonContextProvider = exports.useSkeletonItem_unstable = exports.useSkeletonItemStyles_unstable = exports.skeletonItemClassNames = exports.renderSkeletonItem_unstable = exports.SkeletonItem = exports.useSkeleton_unstable = exports.useSkeletonStyles_unstable = exports.skeletonClassNames = exports.renderSkeleton_unstable = exports.Skeleton = void 0;
var Skeleton_1 = /*#__PURE__*/require("./Skeleton");
Object.defineProperty(exports, "Skeleton", {
  enumerable: true,
  get: function () {
    return Skeleton_1.Skeleton;
  }
});
Object.defineProperty(exports, "renderSkeleton_unstable", {
  enumerable: true,
  get: function () {
    return Skeleton_1.renderSkeleton_unstable;
  }
});
Object.defineProperty(exports, "skeletonClassNames", {
  enumerable: true,
  get: function () {
    return Skeleton_1.skeletonClassNames;
  }
});
Object.defineProperty(exports, "useSkeletonStyles_unstable", {
  enumerable: true,
  get: function () {
    return Skeleton_1.useSkeletonStyles_unstable;
  }
});
Object.defineProperty(exports, "useSkeleton_unstable", {
  enumerable: true,
  get: function () {
    return Skeleton_1.useSkeleton_unstable;
  }
});
var SkeletonItem_1 = /*#__PURE__*/require("./SkeletonItem");
Object.defineProperty(exports, "SkeletonItem", {
  enumerable: true,
  get: function () {
    return SkeletonItem_1.SkeletonItem;
  }
});
Object.defineProperty(exports, "renderSkeletonItem_unstable", {
  enumerable: true,
  get: function () {
    return SkeletonItem_1.renderSkeletonItem_unstable;
  }
});
Object.defineProperty(exports, "skeletonItemClassNames", {
  enumerable: true,
  get: function () {
    return SkeletonItem_1.skeletonItemClassNames;
  }
});
Object.defineProperty(exports, "useSkeletonItemStyles_unstable", {
  enumerable: true,
  get: function () {
    return SkeletonItem_1.useSkeletonItemStyles_unstable;
  }
});
Object.defineProperty(exports, "useSkeletonItem_unstable", {
  enumerable: true,
  get: function () {
    return SkeletonItem_1.useSkeletonItem_unstable;
  }
});
var index_1 = /*#__PURE__*/require("./contexts/index");
Object.defineProperty(exports, "SkeletonContextProvider", {
  enumerable: true,
  get: function () {
    return index_1.SkeletonContextProvider;
  }
});
Object.defineProperty(exports, "useSkeletonContext", {
  enumerable: true,
  get: function () {
    return index_1.useSkeletonContext;
  }
});
