// @fluentui/react-card/lib-amd/components/Card/CardContext.js
define(["require", "exports", "react"], function (require, exports, React) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.useCardContext_unstable = exports.CardProvider = exports.cardContextDefaultValue = void 0;
    var cardContext = React.createContext(undefined);
    /**
     * @internal
     */
    exports.cardContextDefaultValue = {
        selectableA11yProps: {
            referenceId: undefined,
            setReferenceId: function () {
                /* Noop */
            },
            referenceLabel: undefined,
            setReferenceLabel: function () {
                /* Noop */
            },
        },
    };
    /**
     * @internal
     */
    exports.CardProvider = cardContext.Provider;
    /**
     * @internal
     */
    var useCardContext_unstable = function () { var _a; return (_a = React.useContext(cardContext)) !== null && _a !== void 0 ? _a : exports.cardContextDefaultValue; };
    exports.useCardContext_unstable = useCardContext_unstable;
});
