// rmwc/dist/docs/views/app/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.App = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var react_router_1 = require("react-router");
require("../../styles");
var base_1 = require("@rmwc/base");
var top_app_bar_1 = require("@rmwc/top-app-bar");
var icon_1 = require("@rmwc/icon");
var theme_1 = require("@rmwc/theme");
var typography_1 = require("@rmwc/typography");
var ripple_1 = require("@rmwc/ripple");
var drawer_1 = require("@rmwc/drawer");
var list_1 = require("@rmwc/list");
var menu_1 = require("@rmwc/menu");
var base_2 = require("@rmwc/base");
var home_1 = tslib_1.__importDefault(require("../home"));
var site_search_1 = require("../site-search");
var doc_versions_1 = require("../../common/doc-versions");
var menu_content_1 = require("../../common/menu-content");
var theme_picker_1 = require("./theme-picker");
var MainMenuItem = function (_a) {
    var url = _a.url, label = _a.label;
    var location = react_router_1.useLocation();
    return (react_1.default.createElement(list_1.ListItem, { tag: react_router_dom_1.Link, to: url, onClick: function () { return window.scrollTo(0, 0); }, activated: location.pathname === url },
        react_1.default.createElement("span", null, label)));
};
var GithubSvg = function () { return (react_1.default.createElement("svg", { viewBox: "0 0 24 24", width: "24", height: "24" },
    react_1.default.createElement("path", { fill: "currentColor", d: "M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z" }))); };
function AppBar(_a) {
    var onNavClick = _a.onNavClick, children = _a.children;
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(top_app_bar_1.TopAppBar, { fixed: true, className: "app__top-app-bar" },
            react_1.default.createElement(top_app_bar_1.TopAppBarRow, null,
                react_1.default.createElement(top_app_bar_1.TopAppBarSection, { alignStart: true },
                    react_1.default.createElement(top_app_bar_1.TopAppBarNavigationIcon, { onClick: onNavClick, icon: "menu" }),
                    react_1.default.createElement(top_app_bar_1.TopAppBarTitle, { tag: react_router_dom_1.Link, to: "/" }, "RMWC"),
                    react_1.default.createElement(menu_1.SimpleMenu, { handle: react_1.default.createElement("span", { className: "app__version" },
                            react_1.default.createElement("span", null, base_1.RMWC_VERSION),
                            " ",
                            react_1.default.createElement(icon_1.Icon, { icon: "arrow_drop_down" })) },
                        react_1.default.createElement(menu_1.MenuItem, null, base_1.RMWC_VERSION),
                        doc_versions_1.DOC_VERSIONS.map(function (v) { return (react_1.default.createElement(menu_1.MenuItem, { key: v, tag: "a", href: "/version/" + v }, v)); })),
                    react_1.default.createElement(site_search_1.SiteSearch, null)),
                react_1.default.createElement(top_app_bar_1.TopAppBarSection, { alignEnd: true },
                    children,
                    react_1.default.createElement(top_app_bar_1.TopAppBarActionItem, { tag: "a", href: "https://github.com/jamesmfriedman/rmwc", icon: react_1.default.createElement(GithubSvg, null) })))),
        react_1.default.createElement(top_app_bar_1.TopAppBarFixedAdjust, null)));
}
function Nav(props) {
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(drawer_1.Drawer, tslib_1.__assign({ id: "main-nav" }, props),
            react_1.default.createElement(drawer_1.DrawerContent, null,
                react_1.default.createElement(list_1.List, null,
                    react_1.default.createElement(NavItems, { options: menu_content_1.menuContent }))),
            react_1.default.createElement(ripple_1.Ripple, { className: "made-by" },
                react_1.default.createElement(typography_1.Typography, { use: "caption" },
                    react_1.default.createElement(icon_1.Icon, { icon: {
                            icon: 'android-chrome-192x192.png',
                            strategy: 'url'
                        } }),
                    react_1.default.createElement("div", null,
                        react_1.default.createElement("div", null,
                            "Made with",
                            ' ',
                            react_1.default.createElement("span", { role: "img", "aria-label": "heart" }, "\u2764\uFE0F"),
                            ' ',
                            "in Sunny FL.")))))));
}
function App() {
    var location = react_router_1.useLocation();
    var isMobile = window.innerWidth < 640;
    var _a = tslib_1.__read(react_1.useState(!isMobile), 2), menuIsOpen = _a[0], setMenuIsOpen = _a[1];
    var _b = tslib_1.__read(react_1.useState("page-" + (location.pathname.split('/').pop() || 'home')), 2), pageId = _b[0], setPageId = _b[1];
    var _c = tslib_1.__read(react_1.useState(window.localStorage.getItem('rmwcTheme') || 'Baseline'), 2), theme = _c[0], setTheme = _c[1];
    react_1.useEffect(function () {
        isMobile && setMenuIsOpen(false);
        var listener = function (evt) {
            var _isMobile = window.innerWidth < 640;
            if (_isMobile !== isMobile) {
                setMenuIsOpen(!_isMobile);
            }
        };
        window.addEventListener('resize', listener);
        return function () { return window.removeEventListener('resize', listener); };
    }, [isMobile]);
    react_1.useEffect(function () {
        setPageId("page-" + (location.pathname.split('/').pop() || 'home'));
    }, [location]);
    return (react_1.default.createElement(theme_1.ThemeProvider, { options: theme_picker_1.getTheme(theme), className: "app__root", tag: "div", id: pageId },
        react_1.default.createElement(AppBar, { onNavClick: function (evt) {
                setMenuIsOpen(!menuIsOpen);
            } }, !isMobile && (react_1.default.createElement(theme_picker_1.ThemePicker, { selectedThemeName: theme, onThemeClick: function (themeName) {
                window.localStorage.setItem('rmwcTheme', themeName);
                setTheme(themeName);
            } }))),
        react_1.default.createElement("div", { className: "demo-content" },
            react_1.default.createElement(Nav, { open: menuIsOpen, dismissible: !isMobile, modal: isMobile, onClose: function () { return setMenuIsOpen(false); } }),
            react_1.default.createElement(drawer_1.DrawerAppContent, { tag: "main", className: "app__content" },
                react_1.default.createElement(react_router_dom_1.Routes, null,
                    react_1.default.createElement(react_router_dom_1.Route, { key: 'home', path: "/", element: react_1.default.createElement(home_1.default, null) }),
                    getDocRoutes({ options: menu_content_1.menuContent })))),
        react_1.default.createElement(base_2.Portal, null)));
}
exports.App = App;
function NavItems(_a) {
    var options = _a.options;
    var location = react_router_1.useLocation();
    return (react_1.default.createElement(react_1.default.Fragment, null, options.map(function (m) {
        var _a;
        if (m.options) {
            return (react_1.default.createElement(list_1.CollapsibleList, { key: m.label, defaultOpen: m.label === 'Components' ||
                    ((_a = m.options) === null || _a === void 0 ? void 0 : _a.some(function (o) { return o.url && location.pathname === o.url; })), handle: react_1.default.createElement(list_1.SimpleListItem, { text: m.label, metaIcon: "chevron_right" }) },
                react_1.default.createElement(NavItems, { options: m.options })));
        }
        return react_1.default.createElement(MainMenuItem, { label: m.label, url: m.url, key: m.label });
    })));
}
function getDocRoutes(_a) {
    var options = _a.options;
    return options.flatMap(function (value) {
        if (value.options) {
            return getDocRoutes({ options: value.options });
        }
        var Component = function () {
            var Inner = value.component;
            document.title = "RMWC | React Material Web Components | " + value.label;
            return react_1.default.createElement(Inner, null);
        };
        return [
            react_1.default.createElement(react_router_dom_1.Route, { path: value.url, key: value.label, element: react_1.default.createElement(Component, null) })
        ];
    });
}
exports.default = App;
